shared/src/main/scala/mlscript/TypingUnit.scala

```diff
@@ -0,0 +1,155 @@
+package mlscript
+
+import mlscript.utils._, shorthands._
+import scala.collection.immutable.{Map, Set, HashSet}
+
+final case class TypingUnitError(message: String) extends Exception(message)
```

Comment:
I'd rather call that "`IllFormedCodeError`" (and reuse it in more places, though for now it's okay if it's only used here).